practical/ProcessMining/group2/alphaplusminer/alpha_plus_miner.py

```diff
@@ -0,0 +1,479 @@
+from sortedcontainers import SortedSet, SortedDict
+from itertools import chain
+import numpy as np
+from graphviz import Digraph
+
+
+# Alpha Miner plus class
+class AlphaMinerplus:
+    def __init__(self, traces):
+        self.traces = traces  # Traces from an event log
+        self.transitions = SortedSet()
+        self.initial_transitions = SortedSet()
+        self.final_transitions = SortedSet()
+        self.relations = SortedDict()  # Dictionary to keep of track of the relations ->, #, <-, ||
+        self.pairs = []
+        self.maximal_pairs = []
+        self.places = []  # Set of places between maximal pairs
+        self.length_one_loops = None
+
+        # This corresponds to step 3 of the alpha plus algorithm:  3) T' := T \ L1L
+        self.log_without_length_one_loops = None
+        self.F_L1L = None
+        self.W_minusL1L = SortedDict()
+
+
+    def get_transitions(self):
+        # Sets all transitions for the current petri net
+        self.transitions = set(chain.from_iterable(self.traces.values()))
+    
+
+    def getInitialTransitions(self) -> SortedSet:
+        
+        #For each trace get the first activity and add it to the set of initial transitions
+        for index, trace in enumerate(self.traces.values(), start=1):
+         
+            try:
+
+                self.initial_transitions.add(trace[0])
+            except IndexError: # TODO - test this
+            
+                self.initial_transitions = SortedSet()
+        return self.initial_transitions
+
+    def getFinalTransitions(self) -> SortedSet:
```

Comment:
you were using different code formats, most of the time snake_case, here camelCase. Try to align to a common code style within the team for better code quality.